Userland/Libraries/LibJIT/Assembler.cpp

```diff
@@ -1,11 +1,23 @@
 /*
  * Copyright (c) 2023, Andreas Kling <kling@serenityos.org>
+ * Copyright (c) 2023, Nikodem Rabulinski <1337-serenity@nrab.lol>
  *
  * SPDX-License-Identifier: BSD-2-Clause
  */
 
+#include <AK/OwnPtr.h>
 #include <LibJIT/Assembler.h>
+#include <LibJIT/X86_64/Assembler.h>
 
 namespace JIT {
 
+OwnPtr<Assembler> assembler_for_current_arch([[maybe_unused]] Vector<u8>& output)
+{
+#if ARCH(X86_64)
+    return make<X86_64::X86_64Assembler>(output);
+#else
+    return {};
```

Comment:
Could we possibly do a more-descriptive error here at compile time instead, considering we're using the preprocessor here anyways?

Reply:
I could return an error and change the return type to `ErrorOr` but the whole idea was that this function can be called by anyone, regardless if the assembler supports that arch or not, and the caller checks if they got back an Assembler or not

Reply:
If you used `ErrorOr<Assembler>` We could skip the heap indirection. Assuming Assembler remains moveable. Then the compiler could hold an `Optional<Assembler>`

Reply:
Could we though? We'd need to make Assembler not virtual then, no? Which would then mean changing the whole approach